qiime/core/testing/transformer.py

```diff
@@ -0,0 +1,100 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+import collections
+
+from .format import (
+    FourIntsDirectoryFormat,
+    MappingDirectoryFormat,
+    IntSequenceFormat,
+    SingleIntFormat,
+    MappingFormat,
+)
+from .plugin import dummy_plugin
+
+
+@dummy_plugin.register_transformer
+def _1(data: list) -> FourIntsDirectoryFormat:
+    df = FourIntsDirectoryFormat()
+    for i, int_ in enumerate(data, 1):
+        df.single_ints.add(int_, int, num=i)
+    return df
+
+
+@dummy_plugin.register_transformer
+def _2(data: int) -> SingleIntFormat:
+    ff = SingleIntFormat()
+    with ff.open() as fh:
+        fh.write('%d\n' % data)
+    return ff
+
+
+@dummy_plugin.register_transformer
+def _3(df: FourIntsDirectoryFormat) -> list:
```

Comment:
Move these `FourIntsDirectoryFormat` transformers to the bottom, for better viewing experience.